### PR TITLE
Refactor RepositoryHandler and ResolverClient

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs [hunk 1]
```diff
@@ -16,19 +17,22 @@ internal class RepositoryHandler
         private readonly IModelFetcher _modelFetcher;
         private readonly Guid _clientId;
         private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly Uri _repositoryUri;
+        private readonly ResolverClientOptions _clientOptions;
 
-        public Uri RepositoryUri { get; }
```
Comment (azabbasi) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs:20]: The only other place that these properties are accessed is in the ResolverClient that actually passes these down to the RepositoryHandler itself, no need to have backward references as ResolverClient knows about these values itself.

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs [hunk 2]
```diff
@@ -16,19 +17,22 @@ internal class RepositoryHandler
...
-        public ResolverClientOptions ClientOptions { get; }
-
-        public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientdiagnostics, ResolverClientOptions options = null)
+        public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientdiagnostics, ResolverClientOptions options)
         {
-            ClientOptions = options ?? new ResolverClientOptions();
-            RepositoryUri = repositoryUri;
+            Argument.AssertNotNull(options, nameof(options));
```
Comment (azabbasi) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs:25]: Options shouldn't be null and we will not create a new instance if it is, the ResolverClient ctor will take care of it.

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs [hunk 3]
```diff
@@ -16,19 +17,22 @@ internal class RepositoryHandler
...
         private readonly Guid _clientId;
         private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly Uri _repositoryUri;
+        private readonly ResolverClientOptions _clientOptions;
 
-        public Uri RepositoryUri { get; }
-        public ResolverClientOptions ClientOptions { get; }
-
-        public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientdiagnostics, ResolverClientOptions options = null)
+        public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientdiagnostics, ResolverClientOptions options)
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs:23]: `clientDiagnostics` (casing D)

Also, why is this a distinct parameter and not in client options?

Comment (azabbasi) [reply sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs:23]: That is a good question, I haven't seen any other client have the diagnostics as part of the options

**Review by drwill-ms — APPROVED** [review]

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs (hunk 3 above)
Comment (drwill-ms) [reply sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Diagnostics.md

**Review by digimaun — APPROVED** [review]

**Pull request merged**